server/aesdsocket.c

```diff
@@ -0,0 +1,304 @@
+    for(p = servinfo; p != NULL; p = p->ai_next) {
+        if ((sockfd = socket(p->ai_family, p->ai_socktype,
+                p->ai_protocol)) == -1) {
+            perror("server: socket");
+            continue;
+        }
+
+        if (setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &yes,
+                sizeof(int)) == -1) {
+            perror("setsockopt");
+            exit(1);
+        }
+
+        if (bind(sockfd, p->ai_addr, p->ai_addrlen) == -1) {
+            close(sockfd);
+            perror("server: bind");
+            continue;
+        }
+
+        break;
+    }
+
+    freeaddrinfo(servinfo); // all done with this structure
+    if(run_deamon)
+    {
+        pid_t pid;
+        /* create new process */
+        pid = fork ();
+        if (pid == -1)
+        {
+            perror("Fork");
+            return -1;
+        }
+        else if (pid != 0)
+        {
+ 
+            exit (EXIT_SUCCESS);
+        }
+        else
+        {
+            if(setsid()==-1)
+            {
+                perror("Session");
+                return -1;
+            }
+            if(chdir("/")==-1)
+            {
+                perror("Changing directory");
+                return -1;
+            }
+            // redirect fd's 0,1,2 to /dev/null /
+            close(STDIN_FILENO);
+            close(STDOUT_FILENO);
+            close(STDERR_FILENO);
+            open ("/dev/null", O_RDWR); // stdin /
+            dup (0); // stdout /
+            dup (0); // stderror */
+        }
+    }
+
+    if (listen(sockfd, BACKLOG) == -1) {
+        perror("listen");
+        exit(1);
+    }
+
+
+    signal(SIGINT, signal_handler);
+    signal(SIGTERM, signal_handler);
+
+
+    /*addr_size = sizeof their_addr;
+    new_fd = accept(sockfd, (struct sockaddr *)&their_addr, &addr_size);*/
+
+    while(!signal_found) {  // main accept() loop
+
+        sin_size = sizeof their_addr;
+        //int num_of_bytes = 0;
+        new_fd = accept(sockfd, (struct sockaddr *)&their_addr, &sin_size);
+        if (new_fd == -1) {
+            printf("in accept fail with int\n");
+            if(errno == EINTR)
+            {
+                break;
+            }
+            perror("accept");
+            continue;
+        }
+        reading = true;
+        inet_ntop(their_addr.ss_family,
+            get_in_addr((struct sockaddr *)&their_addr),
+            s, sizeof s);
+        printf("server: got connection from %s\n", s);
+
+        int local_fd = open("/var/tmp/aesdsocketdata", O_CREAT | O_RDWR | O_APPEND, S_IRWXU |  S_IRWXG | S_IRWXO );
+        printf("file opened\n");
+        int max_buf_size = BUF_SIZE;
+        char *buf = (char *)malloc(max_buf_size * sizeof(char));
+        int return_idx = 0;
+        int num_of_relloc = 0;
+        bool ret_found = false;
+
+        while ( !ret_found ) //return not found
+        {
+            if(recv(new_fd, buf + (num_of_relloc * BUF_SIZE), BUF_SIZE, 0) == - 1){
```

Comment:
@PeterBraganza I think this code wouldn't be guaranteed to work when `num_of_relloc` was > 0 for a given packet, because it assumes the previous packet filled exactly BUF_SIZE bytes which won't necessarily be the case.